dbt/include/trino/macros/catalog.sql

```diff
@@ -73,9 +137,39 @@
         join columns using ("table_database", "table_schema", "table_name")
         join table_comment using ("table_database", "table_schema", "table_name")
         order by "column_index"
+{%- endmacro %}
 
-  {%- endcall -%}
 
-  {{ return(load_result('catalog').table) }}
+{% macro trino__get_catalog_schemas_where_clause_sql(schemas) -%}
+    where
+        table_schema != 'information_schema'
+        and
+        table_schema in ('{{ schemas | join("','") | lower }}')
+{%- endmacro %}
+
 
+{% macro trino__get_catalog_relations_where_clause_sql(relations) -%}
+    where
+        table_schema != 'information_schema'
+        and
+        (
+            {%- for relation in relations -%}
+                {% if relation.schema and relation.identifier %}
+                    (
+                        table_schema = '{{ relation.schema | lower }}'
+                        and table_name = '{{ relation.identifier | lower }}'
+                    )
+                {% elif relation.schema %}
+                    (
+                        table_schema = '{{ relation.schema | lower }}'
+                    )
+                {% else %}
+                    {% do exceptions.raise_compiler_error(
+                        '`get_catalog_relations` requires a list of relations, each with a schema'
+                    ) %}
+                {% endif %}
+
+                {%- if not loop.last %} or {% endif -%}
```

Comment:
note: the OR currently prevents Trino from optimizing these queries reasonably. 
if this new code is a new to-be-common pattern in DBT, we may want to optimize `information_schema.tables` and `information_schema.columns` processing with this use-case in mind